cpp/collections_module/algorithm/algorithm.cpp

```diff
@@ -0,0 +1,37 @@
+#include "algorithm.hpp"
+
+void Collections::Partition(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::List input_list = arguments[0].ValueList();
+    const int64_t &partition_size = arguments[1].ValueInt();
```

Comment:
I think ValueInt returns new object, so no need for reference here. It is I think reference to temporary